plasma_framework/contracts/mocks/attackers/FailFastReentrancyGuardAttacker.sol

```diff
@@ -0,0 +1,21 @@
+pragma solidity 0.5.11;
+pragma experimental ABIEncoderV2;
+
+import "../../src/utils/FailFastReentrancyGuard.sol";
+
+contract FailFastReentrancyGuardAttacker is FailFastReentrancyGuard {
+
+    event RemoteCallFailed();
+
+    function guardedLocal() public nonReentrant {
```

Comment:
What does this function do?

Reply:
My understanding is that both functions here are just to demonstrate that it would fail

Reply:
Wouldn't this function call itself indefinitely until it runs out of gas?

Reply:
In this implementation of `FailFastReentrancyGuard` it would not. That's a bit like why I choose to use our own one instead of the one from openzepplin so it would fail directly instead of running the function first and then fail.

https://github.com/omisego/plasma-contracts/blob/d9c61822a5359106d87891e87cfa3795e5c80b52/plasma_framework/test/src/utils/FailFastReentrancyGuard.test.js#L10

Reply:
Ok that makes sense.